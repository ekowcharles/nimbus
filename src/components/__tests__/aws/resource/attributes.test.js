import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ResourceAttributes from '../../../aws/resource/attributes'

describe('ResourceAttributes', () => {
  it('renders vpc or subnet description', () => {
    const wrapper = shallow(
      <ResourceAttributes
        resourceAttributes={
          {
            "foo": "bar",
            "list": ["one", "two"],
            "not": undefined
          }
        }
        detailsError={undefined}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})