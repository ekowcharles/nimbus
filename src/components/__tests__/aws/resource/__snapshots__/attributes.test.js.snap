// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceAttributes renders vpc or subnet description 1`] = `
<div
  className="attributes"
>
  <div
    className="details-dropdown"
    onClick={[Function]}
  >
    <div
      className="arrow right"
    />
  </div>
  <div
    className="details"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<ul><li>foo:  <strong>bar</strong></li></ul><ul><li>list</li><ul><ul><li>0:  <strong>o</strong></li></ul><ul><li>1:  <strong>n</strong></li></ul><ul><li>2:  <strong>e</strong></li></ul></ul><ul><ul><li>0:  <strong>t</strong></li></ul><ul><li>1:  <strong>w</strong></li></ul><ul><li>2:  <strong>o</strong></li></ul></ul></ul><ul><li>not:  <strong>undefined</strong></li></ul>",
        }
      }
    />
  </div>
</div>
`;
